File: Directory tree script.py
import os

def mostrar_directorios(ruta, nivel = 0):
    # Verificar si la ruta existe
    if not os.path.exists(ruta):
        print("Directory not found")
        
        return

    # Obtener el nombre del directorio o archivo
    nombre = os.path.basename(ruta)

    # Mostrar el nombre con indentación según el nivel
    print("  " * nivel + "─ " + nombre if os.path.isdir(ruta) else "  " * nivel + "└ " + nombre)

    # Si es un directorio, recorrer su contenido
    if os.path.isdir(ruta):
        for elemento in os.listdir(ruta):
            # Llamada recursiva para mostrar subdirectorios y archivos
            mostrar_directorios(os.path.join(ruta, elemento), nivel + 1)

# Solicitar la ruta al usuario
ruta_usuario = input("Enter directory: ")

# Mostrar el contenido de manera jerárquica
mostrar_directorios(ruta_usuario)

# Mantener la consola abierta
input()

File: Directory tree script.cpp
#include <iostream>
#include <filesystem>
#include <string>

void mostrar_directorios(const std::string& ruta, int nivel = 0)
{
    // Verificar si la ruta existe
    if (!std::filesystem::exists(ruta))
    {
        std::cout << "Directory not found" << std::endl;
        
        return;
    }

    // Obtener el nombre del directorio o archivo
    std::string nombre = std::filesystem::path(ruta).filename().string();

    // Mostrar el nombre con indentación según el nivel
    std::cout << std::string(nivel * 2, ' ') << (std::filesystem::is_directory(ruta) ? "─ " : "└ ") << nombre << std::endl;

    // Si es un directorio, recorrer su contenido
    if (std::filesystem::is_directory(ruta))
    {
        for (const auto& elemento : std::filesystem::directory_iterator(ruta))
        {
            // Llamada recursiva para mostrar subdirectorios y archivos
            mostrar_directorios(elemento.path().string(), nivel + 1);
        }
    }
}

int main()
{
    // Solicitar la ruta al usuario
    std::string ruta_usuario;
    std::cout << "Enter directory: ";
    std::getline(std::cin, ruta_usuario);

    // Mostrar el contenido de manera jerárquica
    mostrar_directorios(ruta_usuario);

    // Mantener la consola abierta
    std::cin.get();

    return 0;
}